packages/api/lambdas/bulk-operation.js

```diff
@@ -118,26 +128,67 @@ function applyWorkflowToGranules({
 async function bulkGranuleDelete(payload) {
   const granuleIds = await getGranuleIdsForPayload(payload);
   const granuleModel = new GranuleModel();
+  const granulePgModel = new GranulePgModel();
   const forceRemoveFromCmr = payload.forceRemoveFromCmr === true;
+  const knex = await getKnexClient({ env: process.env });
   const deletedGranules = [];
   await pMap(
     granuleIds,
     async (granuleId) => {
+      const collectionPgModel = new CollectionPgModel();
+
+      let dynamoGranule;
+      let pgGranule;
+
+      // Try to get the Dynamo record. If it cannot be found, just log
+      // the error and skip it.
       try {
-        const granule = await granuleModel.getRecord({ granuleId });
-        if (granule.published && forceRemoveFromCmr) {
-          await granuleModel.unpublishAndDeleteGranule(granule);
-        } else {
-          await granuleModel.delete(granule);
-        }
-        deletedGranules.push(granuleId);
+        dynamoGranule = await granuleModel.getRecord({ granuleId });
       } catch (error) {
         if (error instanceof RecordDoesNotExist) {
           log.info(`Granule ${granuleId} does not exist or was already deleted, continuing`);
           return;
         }
         throw error;
       }
+
+      // Try to get the PG record. If it cannot be found, ignore it and
+      // move along to unpublishing and deleting only the Dynamo record.
+      // If another error is thrown, throw it here.
+      try {
+        const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
+          knex,
+          deconstructCollectionId(dynamoGranule.collectionId)
+        );
+
+        pgGranule = await granulePgModel.get(knex,
+          {
+            granule_id: granuleId,
+            collection_cumulus_id: collectionCumulusId,
+          });
+      } catch (error) {
+        if (!(error instanceof RecordDoesNotExist)) {
+          throw error;
+        }
+      }
+
+      // Using the Dynamo record as the primary source, unpublish it from
+      // CMR if it's published and we need to force-remove it.
+      let updateResponse;
+
+      if (dynamoGranule && dynamoGranule.published && forceRemoveFromCmr) {
+        updateResponse = await unpublishGranule(knex, dynamoGranule);
```

Comment:
Consider: 
```suggestion
        { pgGranule, dynamoGranule } = await unpublishGranule(knex, dynamoGranule);
```

with the refactor below to just remove the ternary.

Reply:
👍 https://github.com/nasa/cumulus/pull/2108/commits/08b8bc49c34667d535be1d29018301e37c94e13b

This seems nicer. I ALWAYS forget JS's 'destructuring without declaration' syntax with the parens. We have those dynamoGranule/pgGranule vars in the upper scope so this should be a reassignment. Let me know if you had something else in mind!

Reply:
👍🏻  no this looks right.   Nit: I don't think you need the parens, although. that looks updated so it's possible that was sorted anyway.

Reply:
This note is really weird, it's in a commit ref.